packages/fx-core/src/component/driver/file/createOrUpdateDebugProfile.ts

```diff
@@ -0,0 +1,146 @@
+
+  @hooks([
+    addStartAndEndTelemetry(actionName, actionName),
+    updateProgress(getLocalizedString("driver.file.progressBar.createOrUpdateDebugProfile")),
+  ])
+  public async run(
+    args: CreateOrUpdateDebugProfileArgs,
+    context: DriverContext
+  ): Promise<Result<Map<string, string>, FxError>> {
+    return wrapRun(async () => {
+      const result = await this.handler(args, context);
+      return result.output;
+    });
+  }
+
+  @hooks([
+    addStartAndEndTelemetry(actionName, actionName),
+    updateProgress(getLocalizedString("driver.file.progressBar.createOrUpdateDebugProfile")),
+  ])
+  public async execute(
+    args: CreateOrUpdateDebugProfileArgs,
+    ctx: DriverContext
+  ): Promise<ExecutionResult> {
+    let summaries: string[] = [];
+    const outputResult = await wrapRun(async () => {
+      const result = await this.handler(args, ctx);
+      summaries = result.summaries;
+      return result.output;
+    });
+    return {
+      result: outputResult,
+      summaries,
+    };
+  }
+
+  private async handler(
+    args: CreateOrUpdateDebugProfileArgs,
+    context: DriverContext
+  ): Promise<{
+    output: Map<string, string>;
+    summaries: string[];
+  }> {
+    try {
+      let launchSettingsPath = "";
+      if (!args.target) {
+        launchSettingsPath = getAbsolutePath(launchSettingsFilePath, context.projectPath);
+      } else {
+        launchSettingsPath = getAbsolutePath(args.target, context.projectPath);
+      }
+      if (!(await fs.pathExists(launchSettingsPath))) {
+        throw new UserError(
+          "LaunchSettingsFileNotExist",
+          getLocalizedString(
+            "driver.file.createOrUpdateDebugProfile.launchSettingsFileNotExist",
+            launchSettingsPath
+          ),
+          helpLink
+        );
+      }
+      const launchSettingsContent = fs.readFileSync(launchSettingsPath, "utf-8");
+      const data = parse(launchSettingsContent.toString());
+      if (!isCommentObject(data) || !!Array.isArray(data.profiles)) {
+        throw new UserError(
+          "LaunchSettingsFileInvalid",
+          getLocalizedString(
+            "driver.file.createOrUpdateDebugProfile.launchSettingsFileInvalid",
+            launchSettingsPath
+          ),
+          helpLink
+        );
+      }
+      let launchUrl = utils.format(
+        launchUrlTemplate,
+        args.appId,
+        process.env[internalOutputNames.teamsAppTenantId]
+      );
+      if (args.loginHint === undefined || args.loginHint === true) {
+        const tokenObjectRes = await context.m365TokenProvider.getJsonObject({
+          scopes: AppStudioScopes,
+        });
+        if (tokenObjectRes.isErr()) {
+          throw tokenObjectRes.error;
+        }
+        const tokenObject = tokenObjectRes.value;
+        if (tokenObject && "upn" in tokenObject) {
+          launchUrl += `&login_hint=${tokenObject.upn}`;
+        }
+      }
+      (data.profiles as any)[args.name].launchUrl = launchUrl;
```

Comment:
If *{name}* profile not exist, add new profile.

Reply:
Done.

Reply:
For the new profile, does it only contain `launchUrl`?
Prefer to generate a complete profile with command, browser, appurl, ...

Reply:
Add logic to add full profile. Because it's hard to get app type, so I add all content that project may need and users can uncomment part they need.

Reply:
This is to cover more scenarios for existing app so a basic profile is enough. Looks good.